### DISCUSSION TITLE
Typing with explicit hint not used

### DISCUSSION THREAD

**Discussion by guillaume-alliander** [microsoft/pyright#7468] [Q&A]
Why in the following code, `a` is not seen as being of type `LineString`? Pyright does not see any issue there with using `a` as an int.

```python
from typing import cast

import shapely
from shapely.geometry.linestring import LineString

def get_line_string() -> LineString:
    return cast(LineString, shapely.from_wkt("LINESTRING (30 10, 10 30, 40 40)"))

a: int = get_line_string()
b = a + 1
```

Shapely is not properly stubbed, but that's why I thought that using cast and explicit type hint should do the trick. Apparently pyright thinks that the return type of `get_line_string()` is `Any`.

How can I get an error from pyright when I try to type `a` to `int`? I am using shapely as an example here, but I think that this is a generic question.

If I add myself stubs:

```python
# in typings.shapely.__init__.pyi
def from_wkt(geometry:str) -> LineString:  ...
```

Then my editor (vscode/pylance) sees that `from_wkt` does return `LineString`, but the error is still not caught. 

In strict mode, pyright complains that it cannot know the type of `b`, but not that `a` is wrong.

**Comment by erictraut** [comment]
This is because the `shapely` library is not typed and is missing some critical type information. The type `LineString` derives from `BaseGeometry` which derives from `shapely.Geometry` which is an unknown type because it is apparently implemented in a language other than Python, and there is no type stub for it. That means `LineString` effectively inherits from `Any`, which means that a static type checker won't be able to tell whether it derives from `int` or any other class.

If you want to use `shapely` with type checking, you should use type stubs that define its types.

**Reply by guillaume-alliander** [reply]
Thanks. I thought I could lie enough to pyright (with my own stubs) for it not to look higher, but I stand corrected.
